### PR TITLE
prov/efa: Call ibv_wr_read_rdma in rxr_read.c

### PR DESCRIPTION
**Pull request by sunkuamzn**
For SHM EP: use fi_read instead of fi_readmsg

For EFA EP: remove the call to efa_rma_post_read in efa_rma.c

Signed-off-by: Sai Sunku <sunkusa@amazon.com>

### CODE REVIEW

**Review by lurobi — APPROVED** [review]
lgtm.  Perhaps some motivation in the commit message would be helpful though.

**Review by lrbison — APPROVED** [review]

**Review by lrbison — CHANGES_REQUESTED** [review]
CI Failed

File: prov/efa/src/rxr/rxr_read.c [hunk 1]
```diff
@@ -543,30 +536,45 @@ int rxr_read_post_once(struct rxr_ep *ep, struct rxr_read_entry *read_entry,
...
-	msg.context = pkt_entry;
 
 	if (read_entry->lower_ep_type == SHM_EP) {
 		peer = rxr_ep_get_peer(ep, read_entry->addr);
 		assert(peer);
 		assert(peer->is_local && ep->use_shm_for_tx);
-		msg.addr = peer->shm_fiaddr;
-		err = fi_readmsg(ep->shm_ep, &msg, 0);
+		err = fi_read(ep->shm_ep, local_buf, len, desc, peer->shm_fiaddr, remote_buf, remote_key, pkt_entry);
+		goto out;
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr_read.c:545]: use an else branch for EFA_EP instead of `goto`.

File: prov/efa/src/rxr/rxr_read.c [hunk 2]
```diff
@@ -516,22 +516,15 @@ int rxr_read_post_once(struct rxr_ep *ep, struct rxr_read_entry *read_entry,
...
 {
-	struct fi_msg_rma msg = {0};
-	struct iovec iov;
-	struct fi_rma_iov rma_iov;
 	struct rxr_pkt_entry *pkt_entry;
 	struct rdm_peer *peer;
 	struct efa_ep *efa_ep;
+	struct efa_qp *qp;
+	struct efa_conn *conn;
+	struct ibv_sge sge_list[1];
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr_read.c:524]: looks like you just need 1 `sge`  instead of an array

**Review by wzamazon — CHANGES_REQUESTED** [review]

File: prov/efa/src/rxr/rxr_read.c (hunk 2 above)
Comment (sunkuamzn) [reply prov/efa/src/rxr/rxr_read.c]: fixed

**Review by wzamazon — APPROVED** [review]

**Pull request merged**